bokeh/models/mappers.py

```diff
@@ -63,3 +65,229 @@ class LinearColorMapper(ColorMapper):
+    elements, then the resulting colormap will be
+    interpolated between these two values and across
+    all of the colors defined in the palette.  If there
+    are more than two colors, the length must match the
+    number of colors in the palette.  In this configuration
+    each segment value is tied to the palette color of the
+    same index.  This argument is optional.  If it is
+    not provided, the min and max of the data will be used.
+    """)
+
+    alpha = Seq(Float, help="""
+    A python array, the same length as the dictionary with the
+    desired alpha values
+    """
+    )
+
+    interpolationMethod = String(default='linear', help="""
+    A string which describes the method of interpolation.  The default, 'linear',
+    will result in a gradual transition from one color to the next.  Other
+    options are 'step' which will result in a single color for all values 
+    which fall between the segment markers (an effective floor function) 
+    """)
+
+    def __init__(self, palette=None, segments=None, alpha=1, interpolationMethod='linear', **kwargs):
+        # Elegantly handle the possability that some parameters can be passed from the
+        # kwargs structure (because of subclassing)
+        if palette is None and 'palette' in kwargs.keys():
+            palette = kwargs['palette']
+
+        if segments is None and 'segments' in kwargs.keys():
+            segments = kwargs['segments']
+
+        if alpha is None and 'alpha' in kwargs.keys():
+            alpha = kwargs['alpha']
+
+        if interpolationMethod is None and 'interpolationMethod' in kwargs.keys():
+            interpolationMethod = kwargs['interpolationMethod']
+
+
+        # Define a dict to hold any alpha values that we might extract
+        # from the different colors in the passed palette.  These can
+        # come from a 4-tuple or a
+        alpha_hold = {}
+
+        if palette is not None:
+            tmp_palette = expandPalette(palette)
+            palette = tmp_palette['palette']
+            alpha_hold = tmp_palette['alpha_hold']
+            kwargs['palette'] = palette
+
+        if alpha is not None and palette is not None:
+            if not isinstance(alpha, list):
+                # Convert the scalar object to a vector so that the Seq property
+                # type can be used.
+                alpha = [alpha] * len(palette)
+
+            # Overwrite the alpha values with those specified by the colors in the palette.
+            # This may not be the desired functionality.  Who should win, the information in
+            # the palette, or the direct alpha information specified in the alpha parameter?
+            for k,v in list(alpha_hold.items()):
+                alpha[k] = v
+
+            # Check to ensure that all of the alpha values are ok.
+            if any([x > 1 for x in alpha]):
+                raise ValueError('alpha can not have any values greater than 1.0')
```

Comment:
E.g., instead `Seq(Float,...)` you could use `Seq(Percent,...)` and then all this checking is automatically handled:

https://github.com/bokeh/bokeh/blob/master/bokeh/properties.py#L1020